src/tools/doall/src/DOALL_chunking.cpp

```diff
@@ -118,6 +118,103 @@ void DOALL::rewireLoopToIterateChunks(LoopDependenceInfo *LDI) {
+    // step " << *periodicVariableSCC->getStepValue() << "\n";
+
+    /*
+     * Retrieve the relevant values of the periodic variable SCC.
+     */
+    auto initialValue = periodicVariableSCC->getInitialValue();
+    auto period = periodicVariableSCC->getPeriod();
+    auto step = periodicVariableSCC->getStepValue();
+    auto phi =
+        periodicVariableSCC->getPhiThatAccumulatesValuesBetweenLoopIterations();
+    auto taskPHI = cast<PHINode>(task->getCloneOfOriginalInstruction(phi));
+
+    auto entryBlock = (phi->getIncomingValue(0) == initialValue) ? 0 : 1;
+    auto loopBlock = 1 - entryBlock;
+    auto taskLoopBlock =
+        task->getCloneOfOriginalBasicBlock(phi->getIncomingBlock(loopBlock));
+    auto loopValue = phi->getIncomingValue(loopBlock);
+    auto taskLoopValue =
+        task->getCloneOfOriginalInstruction(cast<Instruction>(loopValue));
+
+    /*
+     * Calculate the periodic variable's initial value for the task.
+     * This value is: initialValue + step_size * ((task_id * chunk_size) %
+     * period)
+     */
+    auto coreIDxChunkSize = entryBuilder.CreateMul(task->coreArg,
+                                                   task->chunkSizeArg,
+                                                   "coreIdx_X_chunkSize");
+    auto numSteps = entryBuilder.CreateTrunc(
+        entryBuilder.CreateSRem(coreIDxChunkSize, period, "numSteps"),
+        step->getType());
+    auto numStepsxStepSize = entryBuilder.CreateTrunc(
+        entryBuilder.CreateMul(step, numSteps, "stepSize_X_numSteps"),
+        initialValue->getType());
+    auto chunkInitialValue = entryBuilder.CreateAdd(initialValue,
+                                                    numStepsxStepSize,
+                                                    "initialValuePlusStep");
+    taskPHI->setIncomingValue(entryBlock, chunkInitialValue);
+
+    /*
+     * Determine value of the start of this core's next chunk
+     * from the beginning of the next core's chunk.
+     * Formula: (next_chunk_initialValue + (step_size * (num_cores - 1) *
+     * chunk_size)) % period
+     */
+    auto onesValueForChunking = ConstantInt::get(chunkCounterType, 1);
+    auto chunkStepSize = entryBuilder.CreateTrunc(
+        entryBuilder.CreateMul(entryBuilder.CreateSub(task->numCoresArg,
+                                                      onesValueForChunking,
+                                                      "numCoresMinus1"),
+                               task->chunkSizeArg,
+                               "numCoresMinus1_X_chunkSize"),
+        step->getType());
+    auto chunkStep = entryBuilder.CreateMul(chunkStepSize, step, "chunkStep");
+
+    /*
+     * Add the instructions for the calculation of the next chunk's start value
+     * in the loop's body.
+     */
+    IRBuilder<> loopBuilder(taskLoopBlock);
+    loopBuilder.SetInsertPoint(taskLoopBlock->getTerminator());
+    auto nextChunkValue = loopBuilder.CreateSRem(
+        loopBuilder.CreateAdd(
```

Comment:
Avoid this nested style of programming. It makes your code unnecessary harder to maintain/debug